lib/dialects/sqlite/connection-manager.js

```diff
@@ -44,19 +46,27 @@ class ConnectionManager extends AbstractConnectionManager {
   getConnection(options) {
     options = options || {};
     options.uuid = options.uuid || 'default';
-    options.inMemory = (this.sequelize.options.storage || this.sequelize.options.host || ':memory:') === ':memory:' ? 1 : 0;
+    options.storage = this.sequelize.options.storage || this.sequelize.options.host || ':memory:';
+    options.inMemory = options.storage === ':memory:' ? 1 : 0;
 
     const dialectOptions = this.sequelize.options.dialectOptions;
-    options.readWriteMode = dialectOptions && dialectOptions.mode;
+    options.readWriteMode = dialectOptions && dialectOptions.mode || this.lib.OPEN_READWRITE | this.lib.OPEN_CREATE; // default mode;
 
     if (this.connections[options.inMemory || options.uuid]) {
       return Promise.resolve(this.connections[options.inMemory || options.uuid]);
     }
 
+    if (
+      !options.inMemory &&
+      options.readWriteMode === (this.lib.OPEN_READWRITE | this.lib.OPEN_CREATE)
```

Comment:
Do we only want to provide a folder automatically if the `readWriteMode` is the default? It seems to me, from the names of the options, that we should just check if the `OPEN_CREATE` bit is set, what do you think?

Reply:
I think this is a great idea